File: layout约束动画/testConstant2/testConstant2/ViewController.m
//
//  ViewController.m
//  testConstant2
//
//  Created by Lff on 16/12/2.
//  Copyright © 2016年 Lff. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *chongjiboView;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *topCons;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *right;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *left;


@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.topCons.constant = -300;
//    [self.chongjiboView setTranslatesAutoresizingMaskIntoConstraints:NO];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    [UIView animateWithDuration:2.0 animations:^{
        self.topCons.constant = 300;
        [UIView setAnimationRepeatCount:88];
        [self.view layoutIfNeeded];
    }];
}


@end

File: layout约束动画/testContent/testContent/ViewController.swift
//
//  ViewController.swift
//  testContent
//
//  Created by Lff on 16/12/2.
//  Copyright © 2016年 Lff. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    ///容器
    @IBOutlet weak var contentView: UIView!
    
    ///容器高 Constraint约束
    @IBOutlet weak var contenHeightCons: NSLayoutConstraint!
    
    ///冲击波
    @IBOutlet weak var scanView: UIImageView!

    //高的 Constraint约束
    @IBOutlet weak var scanlineCons: NSLayoutConstraint!

    override func viewDidLoad() {
        super.viewDidLoad()

        self.scanlineCons.constant = -300
    }
    
    override func viewDidAppear(_ animated: Bool) {
        
        self.scanlineCons.constant = 300
        UIView.animate(withDuration: 1.0) {
            UIView.setAnimationRepeatCount(MAXFLOAT)
            self.view.layoutIfNeeded()
        }
    }


}

